api/http/server_test.go

```diff
@@ -66,6 +68,64 @@ func newTestSigner() (*feed.GenericSigner, error) {
 	return feed.NewGenericSigner(privKey), nil
 }
 
+// TestPinUnpinAPI function tests the pinning and unpinning through HTTP API.
+// It does the following
+//    1) upload a file
+//    2) pin file using HTTP API
+//    3) list all the files using HTTP API and check if the pinned file is present
+//    4) unpin the pinned file
+//    5) list pinned files and check if the unpinned files is not there anymore
+func TestPinUnpinAPI(t *testing.T) {
+	// Initialize Swarm test server
+	srv := NewTestSwarmServer(t, serverFunc, nil, nil)
+	defer srv.Close()
+
+	// upload a file
+	data := testutil.RandomBytes(1, 10000)
+	rootHash := uploadFile(t, srv, data)
+
+	// pin it
+	pinMessage := pinFile(t, srv, rootHash)
+
+	// Check if the return message is valid
+	expectedPinMsg := fmt.Sprintf("Address %s pinned", string(rootHash))
+	if string(pinMessage) != expectedPinMsg {
+		t.Fatalf("pin message mismatch, expected %x, got %x", expectedPinMsg, pinMessage)
+	}
+
+	// get the list of files pinned
+	pinnedInfo := listPinnedFiles(t, srv)
+	listInfos := make([]pin.FileInfo, 0)
+	err := json.Unmarshal(pinnedInfo, &listInfos)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if the pinned file is present in the list pin command
+	fileInfo := listInfos[0]
+	if hex.EncodeToString(fileInfo.Address) != string(rootHash) {
+		t.Fatalf("roothash not in list of pinned files")
+	}
+	if !fileInfo.IsRaw {
+		t.Fatalf("pinned file is not raw")
+	}
+	if fileInfo.PinCounter != 1 {
+		t.Fatalf("pin counter is not 1")
+	}
+	if fileInfo.FileSize != uint64(len(data)) {
+		t.Fatalf("data size mismatch, expected %x, got %x", len(data), fileInfo.FileSize)
+	}
+
+	// unpin it
+	unpinMessage := unpinFile(t, srv, rootHash)
+
+	// Check if the return message is valid
+	expectedunPinMsg := fmt.Sprintf("Address %s unpinned", string(rootHash))
+	if string(unpinMessage) != expectedunPinMsg {
+		t.Fatalf("pin message mismatch, expected %x, got %x", expectedunPinMsg, unpinMessage)
+	}
```

Comment:
To add listPinnedFiles check again, to make sure that it works correctly after unpinning?

Reply:
ooh.. Added now.

Reply:
Thanks.